test/aqua.jl

```diff
@@ -0,0 +1,36 @@
+using Test
+using StatsBase
+using Aqua
+
+@testset "Aqua tests (performance)" begin
+    # This tests that we don't accidentally run into
+    # https://github.com/JuliaLang/julia/issues/29393
+    # Aqua.test_unbound_args(StatsBase)
+    ua = Aqua.detect_unbound_args_recursively(StatsBase)
+    @test length(ua) == 0
+
+    # See: https://github.com/SciML/OrdinaryDiffEq.jl/issues/1750
+    # Test that we're not introducing method ambiguities across deps
+    ambs = Aqua.detect_ambiguities(StatsBase; recursive = true)
```

Comment:
Why do we want to test ambiguities recursively? It's generally argued against that in https://github.com/JuliaTesting/Aqua.jl/issues/77 (and against the default setting of testing `Base` and `Core` as well in the Aqua test suite).

Reply:
I figured that it was more thorough.. I'm happy to change this to `ambs = Aqua.detect_ambiguities(StatsBase)`